src/plugins/kibana_utils/public/state_management/url/kbn_url_tracker.ts

```diff
@@ -165,12 +172,19 @@ export function createKbnUrlTracker({
 
     previousActiveUrl = activeUrl;
     activeUrl = getActiveSubUrl(urlWithStates || urlWithHashes);
+    activeUrl = onBeforeNavLinkSaved(activeUrl);
     storageInstance.setItem(storageKey, activeUrl);
   }
 
   function onMountApp() {
     unsubscribe();
     const historyInstance = history || (getHistory && getHistory()) || createHashHistory();
+
+    // set mounted URL as active
+    if (shouldTrackUrlUpdate(historyInstance.location.hash)) {
```

Comment:
Not sure how this is related to the current PR, mind elaborating?

Reply:
I noted that in the description as: 
> This pr also fixes a bug in kbnUrlTracker where state from initial URL wasn't stored in session storage so it wouldn't appear in nav link.

I noticed this bug and needed to fix it to make artificial testing flow working: 

> Bug to validate that the fix works:
Load discover app using a URL with appended fake &searchSessionId=fake
Discover should loaded with an error
Go to dashboard
Check that chrome link to discover app has your URL but without searchSessionId part

In real-world this bug apparently was such a minor thing that no one never noticed

Reply:
How to reproduce this bug without the fix: 

1. Take a kibana URL with some non-default state, open it. 
2. Change app using nav links 
3. Get back to the first app using the nav link. 

Broken behavior: you will land into the default state
Expected (new) behaviour: you should restore your previous state

Reply:
Sorry I missed this comment! :pray: